### DISCUSSION TITLE
UDP stream to VLC

### DISCUSSION THREAD

**Discussion by rcarsey1** [rtl-airband/RTLSDR-Airband#295] [Q&A]
Ideally, I would like to play the stream without buffering delays, etc. which typically occur when rtl_airband encodes and sends to an Icecast server, which in turn sends to a player (usually with its own buffer).

I've been trying to use the new UDP stream feature and have rtl_airband send to a machine which uses VLC to pick it up and play it.   Here seems to be VLC's raw options (the rawaud section)

[https://wiki.videolan.org/VLC_command-line_help/](url)

It would seem that this should work: (i verified that NFM was not compiled into my binary)

`.\vlc --demux=rawaud --rawaud-channels=1 --rawaud-samplerate=8000 --rawaud-fourcc=fl32 udp://@:1234`

Anyhow, if we can figure this out, I'd like to update the wiki because VLC is a staple amonst

**Comment by charlie-foxtrot** [comment]
I haven't used `vlc` but maybe you're missing an IP address in your UDP path (unless it magically uses localhost when nothing is set):

```
udp://[[<source address>]@[<bind address>][:<bind port>]]
                                 UDP stream sent by a streaming server
```

Also, it doesn't work how?  Is the audio bad or no sound at all?

Perhaps use `nc` to verify you can get UDP data from the instance of `rtl_airband` and save it to a file, then use that file to verify you have the `vlc` parameters correct to play the audio, then use `nc` to `vlc` to verify the UDP input parameters are correct.

**Comment by steve1515** [comment]
@rcarsey1 Did you ever get this to work?

I'm trying the same thing on Windows VLC and I can't get it to work. I've confirmed that capturing the UDP traffic into a file and then playing that file works correctly, but VLC directly listening on a UDP socket never works.

Does anyone else know of another player that would work on Windows?

**Reply by steve1515** [reply]
I've done some more testing, but couldn't get VLC to work. I think there might be something wrong with how VLC receives UDP streams, but I'm not exactly sure. I ended up writing my own UDP audio receiver program using the C# NAudio library and it works perfectly with RTLSDR-Airband.

**Reply by Joe392** [reply]
Has anyone had success receiving UDP audio from rtl_airband using a Java based library, perhaps similar to the C# NAudio library?  From what I have seen so far, Java does not natively support a 32 bit audio sample format.  So far I hear 'static' using a simple Java program to receive the audio from rtl_airband.

**Reply by charlie-foxtrot** [reply]
Don't use Java, but if the format is an issue you can use `nc` to pick up the UDP stream then pipe to `sox` to do a format conversion.  For an example see https://github.com/szpajder/RTLSDR-Airband/pull/272#issue-988053185

**Comment by steve1515** [comment]
In case anyone is still looking for an easy way to play the raw audio stream on Windows. I was able to get it to work with FFMpeg's FFPlay utility with the following command line:
`ffplay.exe -nodisp -ac 1 -ar 16000 -analyzeduration 0 -probesize 32 -f f32le -i udp://0.0.0.0:1234`

**Reply by jimrochny** [reply]
In the output section of your rtl_airband .conf file, what did you set  dest_address = " "; to?  Was there a wildcard so any client can connect?

**Reply by steve1515** [reply]
The dest_address is set to the PC's IP where I want the UDP stream sent to. You can't put in a wild card. With UDP you are basically sending to a specific host.
